test/e2e/Tests/Tests/DistributedTracingTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Xunit.Abstractions;
+using Xunit;
+using System.Diagnostics;
+
+namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
+
+[Collection(Constants.FunctionAppCollectionName)]
+public class DistributedTracingTests
+{
+    private readonly FunctionAppFixture _fixture;
+    private readonly ITestOutputHelper _output;
+    private readonly ActivityListener _activityListener;
+
+    public DistributedTracingTests(FunctionAppFixture fixture, ITestOutputHelper testOutputHelper)
+    {
+        _fixture = fixture;
+        _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        _output = testOutputHelper;
+
+        // Initialize the ActivityListener here
+        _activityListener = new ActivityListener
+        {
+            ShouldListenTo = source => source.Name == "DistributedTracingTests",
+            Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllDataAndRecorded,
+            ActivityStarted = activity => { /* Handle activity started */ },
+            ActivityStopped = activity => { /* Handle activity stopped */ }
+        };
+        ActivitySource.AddActivityListener(_activityListener);
+    }
+
+    [Theory]
+    [InlineData("DistributedTracing_HttpStart")]
+    public async Task DistributedTracingTest(string functionName)
+    {
+        // Start Activity
+        ActivitySource activitySource = new ActivitySource("DistributedTracingTests");
+        using Activity? activity = activitySource.StartActivity("HttpTriggerTests");
+
+        Assert.NotNull(activity);
+
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger(functionName, "");
+
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUriAsync(response);
+        var orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetailsAsync(statusQueryGetUri);
+        string output = orchestrationDetails.Output;
+        ActivityContext.TryParse(output, null, out ActivityContext activityContext);
+
+        Assert.Equal(activity?.TraceId.ToString(), activityContext.TraceId.ToString());
+    }
```

Comment:
Are there other tests that are applicable here? Making sure that logs emitted by the function, or internal spans created by the function are all linked correctly?

Reply:
We won't need to check any logs emitted by the function for distributed tracing. We also don't need to check for any internal spans to check if the spans are linked correctly because the trace Id gives us that information. We started an `Activity` in the test and then checked if our Durable Activity function has the same trace Id as the `Activity` that was started in the test (this means that they are linked in the same trace).